vignettes/scoringutils.Rmd

```diff
@@ -0,0 +1,342 @@
+
+## Input formats
+
+Most of the time, the `score()` function will be able to do the entire evaluation for you. All you need to do is to pass in a `data.frame` with the appropriate columns. Which columns are required depends on the format the forecasts come in. The forecast format can either be based on quantiles (see `example_quantile` for the expected format), based on predictive samples (see `example_sample_continuous` and `example_sample_discrete` for the expected format in each case) or in a binary format. The following table gives an overview (pairwise comparisons will be explained in more detail below): 
+
+```{r, echo=FALSE}
+requirements <- data.table(
+  Format = c(
+    "quantile-based", "sample-based", "binary", "pairwise-comparisons"
+  ),
+  `Required columns` = c(
+    "'observed', 'predicted', 'quantile'",
+    "'observed', 'predicted', 'sample'", "'observed', 'predicted'",
+    "additionally a column 'model'"
+  )
+)
+kable(requirements)
+```
+
+Additional columns may be present to indicate a grouping of forecasts. For example, we could have forecasts made by different models in various locations at different time points, each for several weeks into the future. 
+`scoringutils` automatically tries to determine the *unit of a single forecast*, i.e. the combination of existing columns that are able to uniquely identify a single forecast. It uses all existing columns for this, which can sometimes lead to issues. We therefore recommend using the function `set_forecast_unit()` to determine the forecast unit manually. The function simply drops unneeded columns, while making sure that some necessary, 'protected columns' like "predicted" or "observed" are retained. 
+
+```{r}
+colnames(example_quantile)
+
+set_forecast_unit(
+  example_quantile,
+  c("location", "target_end_date", "target_type", "horizon", "model")
+) %>%
+  colnames()
+```
+
+## Constructing a forecast object
+
+The function `as_forecast()` is be used to construct a forecast object and check the input data. It determines the forecast type, creates an object of the appropriate class (`forecast_binary`, `forecast_quantile` or `forecast_sample`), and validates the input data. Objects of class `forecast_*` have a print method that provides additional information. 
+
+```{r}
+head(example_quantile)
+```
+
+```{r}
+forecast_quantile <- example_quantile %>%
+  set_forecast_unit(
+    c("model", "location", "target_end_date", "forecast_date",
+      "target_type", "horizon")
+  ) %>%
+  as_forecast()
+
+forecast_quantile
+```
+
+If you are unsure what your input data should look like, have a look at the `example_quantile`, `example_sample_discrete`, `example_sample_continuous` and `example_binary` data sets provided in the package. 
+
+The output of `as_forecast()` can later be directly used as input to `score()`. This is the recommended workflow (however, `score()` will run `as_forecast()` internally if this hasn't happened before). 
```

Comment:
You're right, the `scoringutils.Rmd` vignette is outdated and given that its structure is not very helpful, we should delete/repalce it. @seabbs wanted to take a look at creating a new one before `2.0.0`. In case he doesn't have time to do that I would just delete it and refer users to the updated version of the paper.